packages/schematics/angular/module/index.ts

```diff
@@ -82,19 +143,42 @@ export default function (options: ModuleOptions): Rule {
     options.name = parsedPath.name;
     options.path = parsedPath.path;
 
+    let routingModulePath: Path | undefined;
+    const isLazyLoadedModuleGen = options.route && options.module;
+    if (isLazyLoadedModuleGen) {
+      options.routingScope = RoutingScope.Child;
+      routingModulePath = getRoutingModulePath(host, options);
+    }
+
     const templateSource = apply(url('./files'), [
-      options.routing ? noop() : filter(path => !path.endsWith('-routing.module.ts.template')),
+      options.routing || isLazyLoadedModuleGen && !!routingModulePath
+        ? noop()
+        : filter(path => !path.endsWith('-routing.module.ts.template')),
       applyTemplates({
         ...strings,
         'if-flat': (s: string) => options.flat ? '' : s,
+        lazyRoute: isLazyLoadedModuleGen,
+        lazyRouteWithoutRouteModule: isLazyLoadedModuleGen && !routingModulePath,
+        lazyRouteWithRouteModule: isLazyLoadedModuleGen && routingModulePath,
         ...options,
       }),
       move(parsedPath.path),
     ]);
+    const { module, ...modulelessOptions } = options;
+    const moduleDasherized = strings.dasherize(options.name);
+    const modulePath =
+      `${!options.flat ? moduleDasherized + '/' : ''}${moduleDasherized}.module.ts`;
 
     return chain([
-      addDeclarationToNgModule(options),
+      !isLazyLoadedModuleGen ? addDeclarationToNgModule(options) : noop(),
+      addRouteDeclarationToNgModule(options, routingModulePath),
       mergeWith(templateSource),
+      isLazyLoadedModuleGen
+        ? schematic('component', {
+            ...modulelessOptions,
```

Comment:
Nit: 
```
{
...options,
module: modulePath,
}
```

Reply:
I think it's safer to keep it with `modulelessOptions` because, if we use spread on `options`, we are overriding `module`. That way, the validity of this statement is dependent on the order which might be implicit to the readers (i.e. if someone switches the positions of these two, for some reason, we will end up with a bug). Anyway, that's my personal perception.